test/test_index_management.py

```diff
@@ -50,12 +56,170 @@ def test_inputs(self):
                 coll.create_search_index(model_kwargs)
 
 
-globals().update(
-    generate_test_classes(
-        _TEST_PATH,
-        module=__name__,
+class TestSearchIndexProse(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        if not os.environ.get("TEST_INDEX_MANAGEMENT"):
+            raise unittest.SkipTest("Skipping index management tests")
+        url = os.environ.get("MONGODB_URI")
+        username = os.environ["DB_USER"]
+        password = os.environ["DB_PASSWORD"]
+        cls.client = MongoClient(url, username=username, password=password)
```

Comment:
Could you add a teardown method to close this client to avoid this traceback in the tests:
```python
[2023/08/07 15:48:48.225] WARNING: found Topology with running threads:
[2023/08/07 15:48:48.225]   Threads: [<PeriodicExecutor(name=pymongo_server_monitor_thread) object at 0x7fc2fef301f0>, <PeriodicExecutor(name=pymongo_server_rtt_thread) object at 0x7fc2fef30ac0>, <PeriodicExecutor(name=pymongo_server_monitor_thread) object at 0x7fc2fef30e80>, <PeriodicExecutor(name=pymongo_server_rtt_thread) object at 0x7fc2fef3d1c0>, <PeriodicExecutor(name=pymongo_server_monitor_thread) object at 0x7fc2fef3daf0>, <PeriodicExecutor(name=pymongo_server_rtt_thread) object at 0x7fc2fef3de20>, <PeriodicExecutor(name=pymongo_srv_polling_thread) object at 0x7fc2fef934f0>]
[2023/08/07 15:48:48.225]   Topology: <Topology <TopologyDescription id: 64d111698762575a88ad80fb, topology_type: ReplicaSetWithPrimary, servers: [<ServerDescription ('dbx-python-lambda-cc527-shard-00-00.nvno8.mongodb.net', 27017) server_type: RSSecondary, rtt: 0.0014610821393405227>, <ServerDescription ('dbx-python-lambda-cc527-shard-00-01.nvno8.mongodb.net', 27017) server_type: RSPrimary, rtt: 0.0008429330310309928>, <ServerDescription ('dbx-python-lambda-cc527-shard-00-02.nvno8.mongodb.net', 27017) server_type: RSSecondary, rtt: 0.0013568061755072462>]>>
[2023/08/07 15:48:48.225]   Creation traceback:
[2023/08/07 15:48:48.225]   File "setup.py", line 192, in <module>
[2023/08/07 15:48:48.225]     setup(
[2023/08/07 15:48:48.225]   File "/opt/python/3.8/lib/python3.8/site-packages/setuptools/__init__.py", line 107, in setup
...
[2023/08/07 15:48:48.225]   File "/data/mci/a334cd896aded2720efd6191a9c67227/src/test/test_index_management.py", line 68, in setUpClass
[2023/08/07 15:48:48.225]     cls.client = MongoClient(url, username=username, password=password)
[2023/08/07 15:48:48.225]   File "/data/mci/a334cd896aded2720efd6191a9c67227/src/pymongo/mongo_client.py", line 832, in __init__
[2023/08/07 15:48:48.225]     self._topology_settings = TopologySettings(
[2023/08/07 15:48:48.225]   File "/data/mci/a334cd896aded2720efd6191a9c67227/src/pymongo/settings.py", line 78, in __init__
[2023/08/07 15:48:48.225]     self._stack = "".join(traceback.format_stack())
[2023/08/07 15:48:48.737] Finished command 'shell.exec' in function 'run tests' (step 2.5 of 2) in 4m27.543475189s.

```

Reply:
Done